configlib/config.go

```diff
@@ -7,12 +7,48 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/metrumresearchgroup/pkgr/rcmd"
 	homedir "github.com/mitchellh/go-homedir"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/afero"
 	"github.com/spf13/viper"
 )
 
+// packrat uses R.version platform, which is not the same as the Platform
+// as printed in R --version, at least on windows
+func packratPlatform(p string) string {
+	switch p {
+	case "x86_64-w64-mingw32/x64":
+		return "x86_64-w64-mingw32"
+	default:
+		return p
+	}
+}
+
+// NewConfig initialize a PkgrConfig passed in by caller
+func NewConfig(cfg *PkgrConfig) {
+	_ = viper.Unmarshal(cfg)
+	if len(cfg.Library) == 0 {
+		rs := rcmd.NewRSettings(cfg.RPath)
+		rVersion := rcmd.GetRVersion(&rs)
+		cfg.Library = getLibraryPath(cfg.Lockfile.Type, cfg.RPath, rVersion.ToFullString(), rs.Platform, cfg.Library)
+	}
+	return
+}
+
+func getLibraryPath(lockfileType, rpath, rversion, platform, library string) string {
+	switch lockfileType {
+	case "packrat":
+		library = filepath.Join("packrat", "lib", packratPlatform(platform), rversion)
+	case "renv":
+		rversion = fmt.Sprintf("R-%s", rversion)
+		library = filepath.Join("renv", "library", rversion, packratPlatform(platform))
```

Comment:
with rEnv the r version is actually different. It just takes `R-3.5` whereas packrat takes 3.5.0 

we should probably switch getLibraryPath to take the rVersion itself then stringify each individually now.

```diff
-library = filepath.Join("packrat", "lib", packratPlatform(platform), rversion)
+library = filepath.Join("packrat", "lib", packratPlatform(platform), rv.ToFullString())
```

for packrat it would need rv.ToFullString() whereas renv needs rv.ToString()

Reply:
how awesome that ToString() exists!

Reply:
haha yeah there were enough places across the r ecosystem that it refers to one or the other so I added both of those from the get-go :-)